File: ls/server/src/main/java/org/opencds/cqf/cql/ls/server/plugin/CommandContribution.java
package org.opencds.cqf.cql.ls.server.plugin;

import java.util.Collections;
import java.util.Set;
import java.util.concurrent.CompletableFuture;
import org.eclipse.lsp4j.ExecuteCommandParams;

public interface CommandContribution {
    default Set<String> getCommands() {
        return Collections.emptySet();
    }

    default CompletableFuture<Object> executeCommand(ExecuteCommandParams params) {
        throw new RuntimeException(String.format("Unsupported Command: %s", params.getCommand()));
    }
}

File: ls/server/src/main/kotlin/org/opencds/cqf/cql/ls/server/plugin/CommandContribution.kt
package org.opencds.cqf.cql.ls.server.plugin

import org.eclipse.lsp4j.ExecuteCommandParams
import java.util.concurrent.CompletableFuture

interface CommandContribution {
    fun getCommands(): Set<String> = emptySet()

    fun executeCommand(params: ExecuteCommandParams): CompletableFuture<Any> {
        throw RuntimeException("Unsupported Command: ${params.command}")
    }
}
